engine/src/ast/function_expr.rs

```diff
@@ -46,24 +43,24 @@ impl<'i, 's, 'a> LexWith<'i, SchemeFunctionParam<'s, 'a>> for FunctionCallArgExp
 
         match ctx.param.arg_kind {
             FunctionArgKind::Field => {
-                let (lhs, input) = LhsFieldExpr::lex_with(input, ctx.scheme)?;
+                let (lhs, input) = IndexExpr::lex_with(input, ctx.scheme)?;
                 if lhs.get_type() != ctx.param.val_type {
                     Err((
                         LexErrorKind::InvalidArgumentType {
                             index: ctx.index,
                             mismatch: TypeMismatchError {
                                 actual: lhs.get_type(),
-                                expected: ctx.param.val_type,
+                                expected: ctx.param.val_type.clone(),
```

Comment:
How come we now need to clone this? It's because `Type` is not copyable anymore?

Reply:
Yes